alternator/executor.cc

```diff
@@ -507,17 +504,21 @@ static rjson::value fill_table_description(schema_ptr schema, table_status tbl_s
             rjson::add(table_description, "GlobalSecondaryIndexes", std::move(gsi_array));
         }
     }
-    // Use map built by describe_key_schema() for base and indexes to produce
-    // AttributeDefinitions for all key columns:
-    rjson::value attribute_definitions = rjson::empty_array();
-    for (auto& type : key_attribute_types) {
-        rjson::value key = rjson::empty_object();
-        rjson::add(key, "AttributeName", rjson::from_string(type.first));
-        rjson::add(key, "AttributeType", rjson::from_string(type.second));
-        rjson::push_back(attribute_definitions, std::move(key));
+    if (tbl_status != table_status::deleting) {
+        // Add base table's KeySchema and collect types for AttributeDefinitions:
+        executor::describe_key_schema(table_description, *schema, key_attribute_types);
+        rjson::add(table_description, "CreationDateTime", rjson::value(creation_date_seconds));
```

Comment:
Please move the setting of CreationDataTime a couple of lines up, the first thing of this if. In the current placement, it's in the middle of the code that handles the schema, so it's confusing.